File: tests/human/x/clj/cross_join_filter.clj
(ns cross-join-filter)

(def nums [1 2 3])
(def letters ["A" "B"])

(def pairs
  (for [n nums
        :when (even? n)
        l letters]
    {:n n :l l}))

(println "--- Even pairs ---")
(doseq [p pairs]
  (println (:n p) (:l p)))

File: tests/human/x/hs/cross_join_filter.hs
module Main where

main :: IO ()
main = do
  let nums = [1,2,3]
      letters = ["A", "B"]
      pairs = [ (n,l) | n <- nums, even n, l <- letters ]
  putStrLn "--- Even pairs ---"
  mapM_ (\(n,l) -> putStrLn $ show n ++ " " ++ l) pairs

